### PR TITLE
[nngpp] update library to latest version (v1.2.4)

### PR DESCRIPTION
**Pull request by ggessner**

### CODE REVIEW

File: ports/nngpp/portfile.cmake [hunk 1]
```diff
@@ -3,8 +3,8 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
```
Comment (LilyWangL) [inline ports/nngpp/portfile.cmake]: `include(vcpkg_common_functions)` is deprecated. Please remove it. And can you please use command `file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

Thanks.

File: ports/nngpp/CONTROL [hunk 2]
```diff
@@ -1,5 +1,5 @@
 Source: nngpp
-Version: 2019-07-25
+Version: nng-v1.2.4
 Homepage:https://github.com/cwzx/nngpp
```
Comment (LilyWangL) [inline ports/nngpp/CONTROL]: Can you please modify this `Homepage:https://github.com/cwzx/nngpp` to `Homepage: https://github.com/cwzx/nngpp`?

Thanks.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/nngpp/CONTROL (hunk 2 above)
Comment (ggessner) [reply ports/nngpp/CONTROL]: thank you very much for your input,
I did not notice that include(vcpkg_common_functions) was in fact deprecated.

File: ports/nngpp/portfile.cmake (hunk 1 above)
Comment (ggessner) [reply ports/nngpp/portfile.cmake]: I made changes to PR accordingly.
Thank you for your input!

**Review by LilyWangL — APPROVED** [review]

File: ports/nngpp/CONTROL [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Source: nngpp
-Version: 2019-07-25
-Homepage:https://github.com/cwzx/nngpp
+Version: nng-v1.2.4
```
Comment (ras0219-msft) [inline ports/nngpp/CONTROL]: Conventionally, we prettify versions like this to just `1.2.4`.

File: ports/nngpp/portfile.cmake [hunk 4]
```diff
@@ -1,10 +1,8 @@
-include(vcpkg_common_functions)
-
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO cwzx/nngpp
-    REF 3351f54e6e774505d8d8b88064d04eb98e0b1cda
-    SHA512 6f72d1085b58ee7a8941294e7479661d8fc2c22cc8af2cee9c2cef11d508032a860c0061851bda07cf995ec8f57e5a25e241a15114a91c487d8aad6def2d4ce5
+    REF nng-v1.2.4
```
Comment (ras0219-msft) [inline ports/nngpp/portfile.cmake]: We prefer to use the git SHA1 here to ensure it cannot change.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
LGTM after my comments are addressed.

**Comment by ras0219-msft** [comment]
Thanks!

**Pull request merged**